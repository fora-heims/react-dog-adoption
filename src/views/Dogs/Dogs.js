import { React, useState, useEffect } from 'react';
import './Dogs.css';
import { getDogs } from '../../services/dogs.js';
import DogList from '../../components/DogList/DogList.js';
import Header from '../../components/Header/Header';

export default function Dogs() {
  const [dogs, setDogs] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const fetchData = async () => {
      const doggiesData = await getDogs();
      setDogs(doggiesData);
      setLoading(false);
    };
    fetchData();
  }, []);

  return (
    <div>
      <Header />
      {loading ? <span>...loading</span> : <DogList dogs={dogs}></DogList>}
    </div>
  );
}
