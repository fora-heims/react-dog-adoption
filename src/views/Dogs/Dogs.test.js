import Dogs from './Dogs.js';
import { MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';

test('', async () => {
  const { container } = render(
    <MemoryRouter>
      <Dogs />
    </MemoryRouter>
  );
  await screen.findByText('Meet Dragon');
  expect(container).toMatchSnapshot();
});
