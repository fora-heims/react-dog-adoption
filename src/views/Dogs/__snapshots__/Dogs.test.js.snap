// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <div>
    <div>
      <header>
        <a
          aria-current="page"
          class="navigate active"
          href="/"
        >
          Home
        </a>
      </header>
    </div>
    <div
      class="list"
    >
      <a
        class="list-item"
        href="/dogs/10"
      >
        <div
          class="detail"
        >
          <h3>
            Meet 
            Barton
          </h3>
          <img
            alt="dog"
            class=""
            src="https://placedog.net/500?id=6"
          />
          <p>
            Barton
             is a(n) 
            1
             year old 
            Weimaraner
          </p>
          <br />
        </div>
      </a>
      <a
        class="list-item"
        href="/dogs/11"
      >
        <div
          class="detail"
        >
          <h3>
            Meet 
            Happy
          </h3>
          <img
            alt="dog"
            class=""
            src="https://placedog.net/500?id=21"
          />
          <p>
            Happy
             is a(n) 
            11
             year old 
            French Bulldog
          </p>
          <br />
        </div>
      </a>
      <a
        class="list-item"
        href="/dogs/14"
      >
        <div
          class="detail"
        >
          <h3>
            Meet 
            Eyes
          </h3>
          <img
            alt="dog"
            class=""
            src="https://placedog.net/500?id=22"
          />
          <p>
            Eyes
             is a(n) 
            4
             year old 
            Weimaraner Mix
          </p>
          <br />
        </div>
      </a>
      <a
        class="list-item"
        href="/dogs/15"
      >
        <div
          class="detail"
        >
          <h3>
            Meet 
            Pugilicious
          </h3>
          <img
            alt="dog"
            class=""
            src="https://placedog.net/500?id=26"
          />
          <p>
            Pugilicious
             is a(n) 
            12
             year old 
            Pug
          </p>
          <br />
        </div>
      </a>
      <a
        class="list-item"
        href="/dogs/5"
      >
        <div
          class="detail"
        >
          <h3>
            Meet 
            Melba
          </h3>
          <img
            alt="dog"
            class=""
            src="https://placedog.net/500?id=8"
          />
          <p>
            Melba
             is a(n) 
            4
             year old 
            Austrailian Cattle Dog
          </p>
          <br />
        </div>
      </a>
      <a
        class="list-item"
        href="/dogs/9"
      >
        <div
          class="detail"
        >
          <h3>
            Meet 
            Liliana
          </h3>
          <img
            alt="dog"
            class=""
            src="https://placedog.net/500?id=5"
          />
          <p>
            Liliana
             is a(n) 
            1
             year old 
            Husky
          </p>
          <br />
        </div>
      </a>
      <a
        class="list-item"
        href="/dogs/6"
      >
        <div
          class="detail"
        >
          <h3>
            Meet 
            Ron
          </h3>
          <img
            alt="dog"
            class=""
            src="https://images.pexels.com/photos/6246667/pexels-photo-6246667.jpeg?auto=compress&cs=tinysrgb&dpr=3&h=750&w=1260"
          />
          <p>
            Ron
             is a(n) 
            less than one
             year old 
            Golden Retriever
          </p>
          <br />
        </div>
      </a>
      <a
        class="list-item"
        href="/dogs/2"
      >
        <div
          class="detail"
        >
          <h3>
            Meet 
            Amely
          </h3>
          <img
            alt="dog"
            class=""
            src="https://placedog.net/500?id=12"
          />
          <p>
            Amely
             is a(n) 
            less than one
             year old 
            Rottweiler
          </p>
          <br />
        </div>
      </a>
      <a
        class="list-item"
        href="/dogs/3"
      >
        <div
          class="detail"
        >
          <h3>
            Meet 
            Althea
          </h3>
          <img
            alt="dog"
            class=""
            src="https://placedog.net/500?id=10"
          />
          <p>
            Althea
             is a(n) 
            1
             year old 
            German Short-haired Pointer
          </p>
          <br />
        </div>
      </a>
      <a
        class="list-item"
        href="/dogs/4"
      >
        <div
          class="detail"
        >
          <h3>
            Meet 
            Franny
          </h3>
          <img
            alt="dog"
            class=""
            src="https://placedog.net/500?id=51"
          />
          <p>
            Franny
             is a(n) 
            4
             year old 
            French Bulldog
          </p>
          <br />
        </div>
      </a>
      <a
        class="list-item"
        href="/dogs/16"
      >
        <div
          class="detail"
        >
          <h3>
            Meet 
            Ginny
          </h3>
          <img
            alt="dog"
            class=""
            src="https://images.pexels.com/photos/5886866/pexels-photo-5886866.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500"
          />
          <p>
            Ginny
             is a(n) 
            less than one
             year old 
            Golden Retriever
          </p>
          <br />
        </div>
      </a>
      <a
        class="list-item"
        href="/dogs/8"
      >
        <div
          class="detail"
        >
          <h3>
            Meet 
            Eileen
          </h3>
          <img
            alt="dog"
            class=""
            src="https://placedog.net/500?id=4"
          />
          <p>
            Eileen
             is a(n) 
            2
             year old 
            American Boxer
          </p>
          <br />
        </div>
      </a>
      <a
        class="list-item"
        href="/dogs/17"
      >
        <div
          class="detail"
        >
          <h3>
            Meet 
            Mrs. Weasley
          </h3>
          <img
            alt="dog"
            class=""
            src="https://placedog.net/1000/563?id=3"
          />
          <p>
            Mrs. Weasley
             is a(n) 
            3
             year old 
            Golden Retriever
          </p>
          <br />
        </div>
      </a>
    </div>
  </div>
</div>
`;
